pkg/utils/parse_arf_result.go

```diff
@@ -17,30 +17,34 @@ const (
 	machineConfigFixType = "urn:xccdf:fix:script:ignition"
 )
 
-func ParseRemediationsFromArf(scheme *runtime.Scheme, scanName string, namespace string, arf string) ([]*complianceoperatorv1alpha1.ComplianceRemediation, error) {
+func ParseRemediationFromContentAndResults(scheme *runtime.Scheme, scanName string, namespace string, content, results string) ([]*complianceoperatorv1alpha1.ComplianceRemediation, error) {
 	remediations := make([]*complianceoperatorv1alpha1.ComplianceRemediation, 0)
 
-	reader := strings.NewReader(arf)
-	dom, err := xmldom.Parse(reader)
+	resultsReader := strings.NewReader(results)
+	resultsDom, err := xmldom.Parse(resultsReader)
 	if err != nil {
 		return nil, err
 	}
 
 	// Get the checks that had failed
-	failedRuleResults := filterFailedResults(dom.Root.Query("//TestResult/rule-result"))
+	failedRuleResults := filterFailedResults(resultsDom.Root.Query("//rule-result"))
 
-	// Get group that contains remediations
-	relevantDom := dom.Root.Query("/report-requests/report-request/content/data-stream-collection/component/Benchmark")
+	dsReader := strings.NewReader(content)
```

Comment:
Same applies here. We could pass a Reader interface for the content.

Reply:
Done locally.